src/Traits/DiffableEntityTrait.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+namespace Equip\Data\Traits;
+
+/**
+ * Intended to be used by classes implementing Equip\Data\EntityInterface that
+ * also need to implement Equip\Data\DiffableInterface.
+ */
+trait DiffableEntityTrait /* implements DiffableInterface */
+{
+    /**
+     * @param array $values
+     *
+     * @return array
+     */
+    public function diff(array $values)
+    {
+        $class = get_class($this);
+        $new = new $class($values);
+        return array_diff_assoc($new->toArray(), $this->toArray());
```

Comment:
Decided on this approach to take advantage of all the logic that `ImmutableValueObjectTrait->apply()`, which is called from `ImmutableValueObjectTrait->__construct()`, already provides to `EntityTrait` (e.g. removing elements from `$values` without corresponding keys in the entity instance).